### DISCUSSION TITLE
Are there any existing tools to stat real-time hit ratio like truenas?

### DISCUSSION THREAD

**Discussion by GrayXu** [openzfs/zfs#14183] [Q&A] [answered]
I know that `arc_summary` reads the stat data of `/proc/spl/kstat/zfs/*` to show overall hit ratio since the last boot (or import). But I was wondering are there any existing tools to stat **real-time** hit ratio like truenas?  

![image](https://user-images.githubusercontent.com/20384421/201911303-ed08799d-beef-44e4-9f1e-3135f6329f84.png)

Or I should write one oneself...

**Comment by GrayXu** [comment, marked as answer]
just found a simple shell version  
CroneKorkN, zfs arc hit ratio monitor, https://gist.github.com/CroneKorkN/476778bb669dcb7610ba7f8ac5a930bb

**Reply by GrayXu** [reply]
found some related discussions  
https://forums.freebsd.org/threads/openzfs-graphing-arc-and-l2arc.83058/  
https://www.reddit.com/r/zfs/comments/436axr/monitoring_zfs_performance_long_term/  
And it seems like many monitoring tools have zfs related plugins

close!

**Comment by grahamperrin** [comment]
Also [sysutils/zfs-stats](https://www.freshports.org/sysutils/zfs-stats/) for FreeBSD. A shot of zfs-mon in xterm: 

![image](https://user-images.githubusercontent.com/192271/204129343-0b80ca95-df04-4c40-99eb-d2f727392d39.png)
